jupytercalpoly/Jupyter-Bifrost
# History Tab MVP

**Added History Functionality**
Fixes #12
Fixes #16

## Changes
- Updated onboarding colors.
- Made global button style.
- Added the SVGs for the clocks on the history tab.
- renamed the `operations_history` variable to `spec_history` since it is more descriptive of storing the specs in Vega format.
- Added the *apply changes* and *export code* buttons.

## Testing
1. Pick a graph in onboarding.
2. Make a change that is visible on the graph.
3. Click *Apply Changes*
4. Go to history.
5. Verify that dragging the slider from left to right toggles between the two graph states
6. Verify that clicking on each description toggles the graph states and the slider changes position.
7. Add a few more points in history, scroll back to the middle of the history and make a change.
8. Press *Apply Changes*
9. Verify that all history past that midpoint was deleted and your current graph was added to the history log.

## Future Work
- Match the styling better, this is just an MVP of the functionality.
- Get an SVG loader to work so that we can use the assets that Angela created.
- Disable the *Apply Changes* button until substantial changes have been made.
- Improve descriptions. While they are decent for mark and column changes, they could be more expressive.
